File: backend/src/main/java/com/technicalescaperoom/backend/service/admin/AdminAuthService.java
package com.technicalescaperoom.backend.service.admin;

import com.technicalescaperoom.backend.entity.AdminSession;
import com.technicalescaperoom.backend.enums.SessionStatus;
import com.technicalescaperoom.backend.repository.AdminSessionRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.time.Instant;
import java.util.UUID;

@Slf4j
@Service
@RequiredArgsConstructor
public class AdminAuthService {

    private final AdminSessionRepository adminSessionRepository;

    @Value("${app.admin.password:admin123}")
    private String adminPassword;

    public String login(String password) {
        if (!adminPassword.equals(password)) {
            throw new IllegalArgumentException("Invalid admin password");
        }

        String sessionToken = UUID.randomUUID().toString() + "-" + Instant.now().toEpochMilli();
        
        AdminSession session = AdminSession.builder()
                .sessionToken(sessionToken)
                .status(SessionStatus.ACTIVE)
                .build();
                
        adminSessionRepository.save(session);
        log.info("Admin successfully logged in. Issued session {}", sessionToken);
        
        return sessionToken;
    }
}

File: backend/services/admin/adminAuthService.js
const { v4: uuidv4 } = require('uuid');
const env = require('../../config/env');
const adminSessionRepository = require('../../repositories/adminSessionRepository');
const adminAuditRepository = require('../../repositories/adminAuditRepository');

class AdminAuthService {
  async login(password) {
    if (!password || password.trim() !== env.ADMIN_PASSWORD) {
      throw new Error('Invalid admin credentials');
    }

    const token = uuidv4();
    await adminSessionRepository.createSession(token);
    await adminAuditRepository.log('admin', 'ADMIN', 'LOGIN', 'ADMIN_PORTAL', 'Admin successfully logged in');
    return token;
  }

  async logout(token) {
    if (token) {
      await adminSessionRepository.terminateSession(token);
      await adminAuditRepository.log('admin', 'ADMIN', 'LOGOUT', 'ADMIN_PORTAL', 'Admin logged out');
    }
  }
}

module.exports = new AdminAuthService();
